demo/grid-styling-demos.html

```diff
@@ -0,0 +1,119 @@
+<dom-module id="grid-styling-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+    <p>Read more about writing custom CSS for themable elements from the 
+      <a href="https://github.com/vaadin/vaadin-themable-mixin/wiki">
+        styling and theming documentation
+      </a>.
+    </p>
+
+    <h3>Generating Class Names for Grid Cells</h3>
+
+    <vaadin-demo-snippet id='grid-styling-demos-cell-class-generator'>
+      <template preserve-content>
+
+        <vaadin-grid aria-label="Styling Grid With Cell Class Generator Example" id="grid" size="200">
+          <vaadin-grid-column path="name.first" header="First name"></vaadin-grid-column>
+          <vaadin-grid-column path="name.last" header="Last name"></vaadin-grid-column>
+          <vaadin-grid-column path="email"></vaadin-grid-column>
+        </vaadin-grid>
+
+        <script>
+          window.addDemoReadyListener('#grid-styling-demos-cell-class-generator', function(document) {
+            const grid = document.querySelector('vaadin-grid');
+            grid.items = Vaadin.GridDemo.users;
+
+            grid.cellClassNameGenerator = function(rowData, column) {
+              let classes = rowData.item.gender;
+              if (column.path === 'email') {
+                classes += ' email';
+              } else if (column.path === 'name.first' && rowData.item.name.title === 'mrs') {
+                classes += ' mrs';
+              }
+              return classes;
+            };
+          });
+        </script>
+
+        <!-- Place in index.html or in a separate HTML import -->
+        <dom-module id="my-grid-styles" theme-for="vaadin-grid">
+          <template>
+              <style>
+                /* Background needs a stronger selector to not be overridden */
+                [part~="cell"].female {
+                    background: var(--lumo-primary-color-10pct);
+                }
+
+                [part~="cell"].male {
+                  background: var(--lumo-contrast-5pct);
+                }
+
+                .email {
+                  font-style: italic;
+                }
+
+                .mrs {
+                  font-weight: bold;
+                }
+              </style>
+          </template>
+        </dom-module>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Updating Styles After Changes in the Data</h3>
+
+    <vaadin-demo-snippet id='grid-styling-demos-regenerate-classes'>
+      <template preserve-content>
+
+        <vaadin-button style="margin-bottom: 20px" id="change-title-btn">Change Title</vaadin-button>
```

Comment:
Just looking at the demo, it's a bit unclear that clicking a global "Change Title" button changes the marital status of the first item in the grid.

Reply:
Fixed: Now the demo has a column of checkboxes to change the styles on any row.